src/reducers/bookmarkArticleReducer.js

```diff
@@ -0,0 +1,44 @@
+import initialState from '../store/initialState';
+import { bookmarkArticleType } from '../actions/bookmarkActions';
+
+/**
+ *
+ * @param {*} state - The initial bookmarkedList state
+ * @param {*} action - The action payload
+ * @returns {Object} - The current bookmarkedList state
+ */
+const bookmarkReducer = (state = initialState.bookmarkedList, action) => {
+  const { type, data, message } = action;
+  switch (type) {
+    case bookmarkArticleType.success:
+      if (message === 'successfully unbookmarked this article') {
+        const filteredList = state.bookmarked.filter(post => post.articleId !== data.articleId);
```

Comment:
Thank you for this observation. The current implementation of the bookmark feature on the backend requires me to do this. The current implementation uses the same route and method to bookmark and unbookmark an article, so both cases are successful cases.

Also, importantly, note that while `getState` will fetch the current state, it does not mutate the state, and since the bookmarked article has been saved in the store, we need to be able to remove the bookmarked article from the store.